internal/locate/region_request.go

```diff
@@ -617,6 +646,14 @@ func (state *accessFollower) next(bo *retry.Backoffer, selector *replicaSelector
 				zap.Bool("leader-invalid", leaderInvalid),
 				zap.Any("labels", state.option.labels))
 		}
+		// If deadline exceeded error count is greater than or equal to the number of replicas,
+		// return nil to upper layer to retry with default timeout.
+		if state.deadlineExceededErrCnt >= len(selector.replicas) {
+			return nil, nil
+		}
+		if state.deadlineExceededErrCnt >= 2 && state.isStaleRead {
```

Comment:
If the first try receives data-is-not-ready error and second try receives a deadline error, it'll fallback to `tryFollower` and the slow replica may be tried again in it.

This question is about should we mark the deadline exceeded replicas? Because state change is allowed in replica selector and the slow replica may be avoided when state is changed.

Reply:
nice catch, I will refactor this.